config/initializers/preview_generation.rb

```diff
@@ -0,0 +1,11 @@
+ActiveStorage::Previewer::VideoPreviewer.class_eval do
+  def draw_relevant_frame_from(file, &block)
+    draw self.class.ffmpeg_path, "-i", file.path, *video_preview_arguments_with_mid_video_start_time, "-", &block
+  end
+
+  def video_preview_arguments_with_mid_video_start_time
+    # Ensures that the blob has been analysed, otherwise there will be no duration and it will default to the start of the video
+    blob.analyze
```

Comment:
Just a note to myself that I checked this out to make sure that I understood what this will do.
This could potentially add a little time to the preview, since it synchronously analyses the file (and I think downloads it again). We could optimise this more by adding a condition - `blob.analyze unless blob.metadata[:analyzed]`. 

Since previews are cached as their own blob, we probably don't need to worry too much about being super optimised here though.